tools/data/ava/preparing_ava.md

```diff
@@ -0,0 +1,136 @@
+# Preparing AVA
+
+For basic dataset information, please refer to the official [website](https://research.google.com/ava/index.html).
+Before we start, please make sure that the directory is located at `$MMACTION2/tools/data/ava/`.
+
+## Step 1. Prepare Annotations
+
+First of all, you can run the following script to prepare annotations.
+
+```shell
+bash download_annotations.sh
+```
+
+This command will download `ava_v2.1.zip` for AVA `v2.1` annotation. If you need the AVA `v2.2` annotation, you can try the following script.
+
+```shell
+VERSION=2.2 bash download_annotations.sh
+```
+
+## Step 2. Prepare Videos
+
+Then, use the following script to prepare videos. The codes are adapted from the [official crawler](https://github.com/cvdfoundation/ava-dataset).
+Note that this might take a long time.
+
+```shell
+bash download_videos.sh
+```
+
+Or you can use the following command to downloading AVA videos in parallel using a python script.
+```shell
+bash download_videos_parallel.sh
+```
+
+Note that if you happen to have sudoer or have [GNU parallel](https://www.gnu.org/software/parallel/) on your machine,
+you can speed up the procedure by downloading in parallel.
+
+```shell
+# sudo apt-get install parallel
+bash download_videos_gnu_parallel.sh
+```
+
+## Step 3. Cut Videos
+
+Cut each video from its 15th to 30th minute and make them at 30 fps.
+
+```shell
+bash cut_videos.sh
+```
+
+## Step 4. Extract RGB and Flow
+
+This part is **optional** if you only want to use the video loader.
+
+Before extracting, please refer to [install.md](/docs/install.md) for installing [denseflow](https://github.com/open-mmlab/denseflow).
+
+If you have plenty of SSD space, then we recommend extracting frames there for better I/O performance. And you can run the following script to soft link the extracted frames.
+
+```shell
+# execute these two line (Assume the SSD is mounted at "/mnt/SSD/")
+mkdir /mnt/SSD/ava_extracted/
+ln -s /mnt/SSD/ava_extracted/ ../data/ava/rawframes/
+```
+
+If you only want to play with RGB frames (since extracting optical flow can be time-consuming), consider running the following script to extract **RGB-only** frames using denseflow.
+
+```shell
+bash extract_rgb_frames.sh
+```
+
+If you didn't install denseflow, you can still extract RGB frames using OpenCV by the following script, but it will keep the original size of the images.
+
+```shell
+bash extract_rgb_frames_opencv.sh
+```
+
+Or using ffmpeg to extract RGB frames by the following script.
+
+```shell
+bash extract_rgb_frames_ffmpeg.sh
```

Comment:
Since we provide 2 approaches for frame extraction, ideally, they should extract frames in the same fashion. However, I noticed that when using ffmpeg, the framerate was specified to be 30; when using opencv, the framerate was not specified. The 2 approaches are the same only if all videos in AVA are 30fps, I'm not sure if this is the case.

Reply:
`cut_videos.sh` will use `ffmpeg` to make the video 30 FPS. (`-r 30`)

Reply:
That would be fine.